src/universal/components/AuthProvider.tsx

```diff
@@ -0,0 +1,17 @@
+import {Component} from 'react'
+
+class AuthProvider extends Component<{}> {
+  componentDidMount () {
+    const params = new URLSearchParams(window.location.search)
+    const state = params.get('state')
+    const code = params.get('code')
+    if (!window.opener) return
+    window.opener.postMessage({state, code})
```

Comment:
here's a fun new pattern to pass back the code from oauth. we send back the code from the window that opened the popup.

Reply:
👯